### ISSUE TITLE
What is the objective function to optimize?

### ISSUE THREAD

**Issue by xywust2014** [dreamquark-ai/tabnet#140]
I notice that for every epoch, there will be train and valid accuracy. 
Is the accuracy the metrics for the optimization? I am currently dealing with binary classification problem, and I would like to use auc or recall as an metric. May I be able to do that too? 

Thank yo very much for your response.

**Comment by Optimox** [comment]
hello @xywust2014,

There are two distinct metrics to consider:
- the loss function that your model is trained to minimize
- the early stopping metric, which is used to select the best epoch and stop the training when no further improvement has been seen.

Currently you can easily change the loss function by using `loss_fn` parameter during the fit, you can pick any loss function from pytorch https://pytorch.org/docs/stable/nn.html or even create your own if you like. By default TabNetClassifier uses cross entropy while TabnetRegressor uses Mean Squarred Error.

About the early stopping metric, you can't change it easily with the current implementation, this will be improved in the future. For binary classification the early stop metric (which is the metric that the training displays) is AUC. For multiclass classfication it's the accuracy. For regression it's the mean squarred error.

You can easily access your training loss and early stop metrics by calling `clf.history['train']['loss']` or `clf.history['train']['metric']` or with the same command for the valid set `clf.history['valid']['metric']`

Hope this helps!

**Comment by xywust2014** [comment]
> hello @xywust2014,
> 
> There are two distinct metrics to consider:
> 
> * the loss function that your model is trained to minimize
> * the early stopping metric, which is used to select the best epoch and stop the training when no further improvement has been seen.
> 
> Currently you can easily change the loss function by using `loss_fn` parameter during the fit, you can pick any loss function from pytorch https://pytorch.org/docs/stable/nn.html or even create your own if you like. By default TabNetClassifier uses cross entropy while TabnetRegressor uses Mean Squarred Error.
> 
> About the early stopping metric, you can't change it easily with the current implementation, this will be improved in the future. For binary classification the early stop metric (which is the metric that the training displays) is AUC. For multiclass classfication it's the accuracy. For regression it's the mean squarred error.
> 
> You can easily access your training loss and early stop metrics by calling `clf.history['train']['loss']` or `clf.history['train']['metric']` or with the same command for the valid set `clf.history['valid']['metric']`
> 
> Hope this helps!

Thank you! Optimox. This helps a lot.
